lxd/containers.go

```diff
@@ -159,6 +292,122 @@ type containerStatePutReq struct {
 	Force   bool   `json:"force"`
 }
 
+type lxdContainer struct {
+	c *lxc.Container
+	id int
+}
+
+func (c *lxdContainer) Start() error {
+	return c.c.Start()
+}
+
+func (c *lxdContainer) Reboot() error {
+	return c.c.Reboot()
+}
+
+func (c *lxdContainer) Freeze() error {
+	return c.c.Freeze()
+}
+
+func (c *lxdContainer) Shutdown(timeout time.Duration) error {
+	return c.c.Shutdown(timeout)
+}
+
+func (c *lxdContainer) Stop() error {
+	return c.c.Stop()
+}
+
+func (c *lxdContainer) Unfreeze() error {
+	return c.Unfreeze()
+}
+
+func newLxdContainer(name string, daemon *Daemon) (*lxdContainer, error) {
+	d := &lxdContainer{}
+
+	certdbname := shared.VarPath("lxd.db")
+	db, err := sql.Open("sqlite3", certdbname)
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+	rows, err := db.Query("select id, architecture from containers where name = ?", name);
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	d.id = -1
+	arch := 0
+	for rows.Next() {
+		var id int
+		rows.Scan(&id, &arch)
+		d.id = id
+	}
+	if d.id == -1 {
+		return nil, fmt.Errorf("Unknown container")
+	}
+
+	c, err := lxc.NewContainer(name, daemon.lxcpath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.SetConfigItem("lxc.include", "/usr/share/lxc/config/ubuntu.common.conf")
```

Comment:
Probably fine for now as everything is hardcoded, but moving forward, I think we'll want to have LXD only include /usr/share/lxc/config/userns.common.conf (if security.privileged == false) or /usr/share/lxc/config/common.conf (if security.privileged == true).

Possibly having an entry in the image which defines additional includes, though I'm not particularly fond of that idea since it then ties the container image to a given release of LXC. Instead I'd argue we should just keep on working upstream to reduce the configuration delta between templates and get to the point where including only common.conf or userns.common.conf is enough.